homestar-runtime/src/event_handler/swarm_event.rs

```diff
@@ -93,14 +97,174 @@ async fn handle_swarm_event<THandlerErr: fmt::Debug + Send, DB: Database>(
     event_handler: &mut EventHandler<DB>,
 ) {
     match event {
-        // TODO: add identify for adding compatable kademlia nodes.
         // TODO: use kademlia to discover new gossip nodes.
+        SwarmEvent::Behaviour(ComposedEvent::Identify(identify_event)) => {
+            match identify_event {
+                identify::Event::Error { peer_id, error } => {
+                    error!(err=?error, peer_id=peer_id.to_string(), "error while attempting to identify the remote")
```

Comment:
check: we can remove these `to_strings` if display is already implemented for peer_id?